### PR TITLE
Feat/arduino pressure monitor

### PR DESCRIPTION
**Pull request by hiromon0125**
# Feature
* Arduino will turn on the compressor and pressurize until max pressure is reached
* Arduino will also serial write the pressure it is reading
* Arduino will open the valve to bend the leg when only the valve button is pressed
* Arduino will also close the valve if the valve is opened for 5 seconds

# How to Test
* upload the code using the arduino IDE if not uploaded on the arduino already
* Check to make sure the following is satisfied
  - The valve next to the compressor is closed
  - The leg is properly attached
* Turn on the board using the big switch
* Open the serial monitor (pressure reading should be seen)
* Wait for the system to pressurize
* Once pressurized check to make sure 82 psi is reached
* Press on the valve button
* the muscle should inflate 
* the compressor should turn back on if the lower than ideal pressure
* Once the muscle is inflated for 5 seconds the valve will close
* pressing on the valve switch past this point will cause the muscle to inflate again
* When the muscle is inflated and both the compressor button and the valve switch is pressed then the valve should close 
* Only pressing on the compressor button will cause the compressor to turn and overwrite the system

merged with #18

### CODE REVIEW

File: src/arduino/arduino.ino [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+#define SENSOR_PIN A0         // Analog
+#define COMP_SWITCH_PIN 11    // button #1
+#define VALVE_1_SWITCH_PIN 10 // button #2
+
+// constants
+#define VOLTAGE_CONST 5
+#define P_MAX 145 // psi
+#define P_MIN 0
+
+#define SYSTEM_PRESSURE 82    // average pressure we want to maintain
```
Comment (cvr8924) [inline src/arduino/arduino.ino:18]: What are the units here? PSI? Cause below this is set to 10 but its implied that we want to have it be 3 psi. Might wanna clariffy documentation here

File: src/arduino/components/Pressure.cpp [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+ * LOGIC:
+ * 1. Read sensor value
+ * 2. Convert sensor value to voltage
+ * 3. Convert voltage to pressure
+ * 4. Set pressure
+ * 5. Return pressure
+ */
+float Pressure::getPressure()
+{
+  double sensorRead = analogRead(sensorPin) * 0.0049;                                                                // Volts
```
Comment (cvr8924) [inline src/arduino/components/Pressure.cpp:15]: whats the magic number here?

File: src/arduino/components/Pressure.cpp [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+  double pressure = (((sensorRead - 0.1 * VOLTAGE_CONST) * (P_MAX - P_MIN)) / (0.8 * VOLTAGE_CONST)) + P_MIN + 0.36; // PSI
```
Comment (cvr8924) [inline src/arduino/components/Pressure.cpp:16]: Again we are pulling constants out of nowhere

File: src/arduino/components/Pressure.cpp [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+/**
+ * Turns compressor on or off
+ *
+ * LOGIC:
+ * if pressure is less than IDEA_PRESSURE or override is true:
+ *   turn on compressor
+ * else:
+ *   turn off compressor
+ */
+void Pressure::Pressurize(bool override)
```
Comment (cvr8924) [inline src/arduino/components/Pressure.cpp:38]: I'd love to see some clarity on when override gets used here

**Review by cvr8924 — APPROVED** [review]
I'm gonna tenitively approve here. I have a few concerns I wanna hash out but I don't think any of them are so significant this cant go to dev branch

File: src/arduino/components/Pressure.cpp (hunk 2 above)
Comment (hiromon0125) [reply src/arduino/components/Pressure.cpp:15]: I think its noise of the pin but I have no idea

File: src/arduino/components/Pressure.cpp (hunk 3 above)
Comment (hiromon0125) [reply src/arduino/components/Pressure.cpp:16]: tbh its dev's code so I also have no clue

File: src/arduino/components/Pressure.cpp (hunk 4 above)
Comment (hiromon0125) [reply src/arduino/components/Pressure.cpp:38]: override is used when the switch is pressed and cases like that

File: src/arduino/components/Pressure.cpp (hunk 2 above)
Comment (hiromon0125) [reply src/arduino/components/Pressure.cpp:15]: my guess based on the equation the voltage of the analog signal is 1/200 of the actual value of the sensor.

File: src/arduino/components/Pressure.cpp (hunk 3 above)
Comment (hiromon0125) [reply src/arduino/components/Pressure.cpp:16]: https://forum.digikey.com/t/topic/7152
according to this japanese guy the equation used is 
(Vread - Vout_min) * (pressure_range) / (Vout_max - Vout_min)
So reverse engineering the constants this is what I get
pressure_range = 145
Vout_max = 4.5
Vout_min = 0.5
There is also the last value of 0.36 I think that is just some calibration to make sure the 0 psi is regular room pressure of 14.7psi

File: src/arduino/components/Pressure.cpp (hunk 4 above)
Comment (hiromon0125) [reply src/arduino/components/Pressure.cpp:38]: You can also see it being used in the loop function

File: src/arduino/components/Pressure.cpp (hunk 2 above)
Comment (hiromon0125) [reply src/arduino/components/Pressure.cpp:15]: I asked Ryland and he pointed to this document https://docs.google.com/document/d/1SdptUUIz2HHxYaZK3It0fdqb0d3JrSSpMELT_E1a4ds/edit

**Pull request closed without merging**